### PR TITLE
km_opt=5 heat flux fix should just use lowest level qv

### PR DESCRIPTION
**Pull request by dudhia**
TYPE: bug fix

KEYWORDS: km_opt=5, 3d tke scheme, use_theta_m=1

SOURCE: internal

DESCRIPTION OF CHANGES:
Problem:
A previous surface flux fix (61029b1416dfb1657) applied a theta_m correction to all levels.  However, this should only 
be used for qv at the surface.

Solution:
Switch k to kts in one place where this is applied, and pull the invariant assignment out of the k-index DO loop.

LIST OF MODIFIED FILES: 
M     dyn_em/module_diffusion.F

TESTS CONDUCTED: 
1. Jenkins testing is all pass.

RELEASE NOTE: Addendum to heat-flux theta_m fix (61029b1416dfb1657)

### CODE REVIEW

File: phys/module_sf_noahmpdrv.F [hunk 1]
```diff
@@ -3206,7 +3206,8 @@ SUBROUTINE noahmp_urban(sf_urban_physics,   NSOIL,         IVGTYP,  ITIMESTEP,
     qfx(i,j)      = (1.-frc_urb2d(i,j))*qfx_rural(i,j)   + qfx_urb(i,j)
     lh(i,j)       = qfx(i,j)*xlv
     hfx(i,j)      = hfx_urb(i,j)                         + (1-frc_urb2d(i,j))*hfx_rural(i,j)      ![W/m/m]
-    sh_urb2d(i,j) = hfx_urb(i,j)/frc_urb2d(i,j)
+!   sh_urb2d(i,j) = hfx_urb(i,j)/frc_urb2d(i,j)
```
Comment (davegill) [inline phys/module_sf_noahmpdrv.F]: We should just remove this commented out line.

**Comment by dudhia** [comment]
Just fixed it.
That was a different fix I did not want in here.
Should be 1 change.

File: dyn_em/module_diffusion_em.F [hunk 2]
```diff
@@ -8107,7 +8107,7 @@ SUBROUTINE vertical_diffusion_implicit(ru_tendf, rv_tendf, rw_tendf, rt_tendf,&
           DO  k = kts+1, ktf-1
             rdz_w = -g/dnw(k)/(c1h(k)*mu(i,j) + c2h(k))
             IF(config_flags%use_theta_m == 1)THEN
-                qfac = 1.+rvovrd*moist(i,k,j,P_QV)
+                qfac = 1.+rvovrd*moist(i,kts,j,P_QV)
```
Comment (davegill) [inline dyn_em/module_diffusion_em.F:8110]: Jimy,
Now that is this not a function of the k index, this qfac assignment should be pulled out of the inner k-loop.

**Comment by dudhia** [comment]
OK, will do.

**Comment by davegill** [comment]
@weiwangncar 
Wei,
I am good with the PR. It is a narrowly focussed change and the reg tests are OK. Jimy pulled an invariant assignment out of the DO loop, which was my only request.  I tweaked the PR message to include the specific previous commit that introduced this problem.

**Review by weiwangncar — APPROVED** [review]

**Pull request merged**